### ISSUE TITLE
useSupabaseSession (.value) in middleware is null when landing on page directly

### ISSUE THREAD

**Issue by franzwarning** [nuxt-modules/supabase#349]
When doing client side transitions, accessing the .value of useSupabaseSession works perfectly. When hitting the page directly, the session.value is null when it shouldn't be. However, the session object looks ok...

```
middlewaresession: ObjectRefImpl {
  _object:
   { '$ssupabase_user':
      { id: '4f5046ab-c291-4519-9415-89efe53e6951',
        aud: 'authenticated',
        role: 'authenticated',
        email: 'xxx',
        email_confirmed_at: '2024-04-11T23:09:56.171727Z',
        phone: '',
        confirmed_at: '2024-04-11T23:09:56.171727Z',
        last_sign_in_at: '2024-04-13T11:14:43.939754Z',
        app_metadata: [Object],
        user_metadata: {},
        identities: [Array],
        created_at: '2024-04-11T23:09:56.16799Z',
        updated_at: '2024-04-13T11:14:43.940623Z' },
     '$s__nuxt_devtools__': { timeSsrStart: 1713009709216 },
     '$ssupabase_session': null },
  _key: '$ssupabase_session',
  _defaultValue: undefined,
  __v_isRef: true }

```

any ideas here?

**Comment by vinceglb** [comment]
I faced the same issue.

I tried to reproduce using the playground in this repository without success.
But I tried to remove modifications done in this commit https://github.com/nuxt-modules/supabase/pull/338#issuecomment-2026938616 and I reproduce the behavior.

Also, the issue https://github.com/nuxt-modules/supabase/issues/350 makes me think that the commit https://github.com/nuxt-modules/supabase/commit/9dee37ee030506fc7888dfbb6cc1f29b9126a728 is not shipped in the 1.2.0.

@larbish, are you sure to ship this https://github.com/nuxt-modules/supabase/commit/9dee37ee030506fc7888dfbb6cc1f29b9126a728 in the latest version?

**Comment by larbish** [comment]
Everything has been shipped in `v1.2.1` (including your improvement @vinceglb), could you have a test ?
